app/controllers/settings/import_controller.rb

```diff
@@ -34,9 +34,11 @@ def myanimelist
 
     mixpanel.track 'Imported from MyAnimeList', {email: current_user.email} if Rails.env.production?
   rescue Exception
-    error! 500, 'There was a problem importing your list.  Please send an email
-                 to josh@hummingbird.me with the file you are trying to import
-                 and we\'ll see what we can do.'
-     raise
+    status = User.import_statuses[:error]
+    current_user.update_columns import_status: status,
+      import_error: 'There was a problem importing your list. Please send an email
+                    to josh@hummingbird.me with the file you are trying to import
+                    and we\'ll see what we can do.'
+    raise
```

Comment:
So basically if something went wrong before the Sidekiq task was created, it would never reset them to an unerrored state?

Damn, I looked at this code like 30 times trying to find the hole and never noticed that

We should probably still `error!` to pitch it down to the client immediately though, right?